reporter/registry.go

```diff
@@ -0,0 +1,103 @@
+package reporter
+
+import (
+	"sync"
+
+	metrics "github.com/rcrowley/go-metrics"
+	"github.com/wavefronthq/wavefront-sdk-go/histogram"
+)
+
+type WavefrontRegistry struct {
```

Comment:
I don't think we need this struct. It seems redundant compared to https://github.com/rcrowley/go-metrics/blob/master/registry.go. The base registry also uses locks and maintains an internal map. So essentially we are double locking and using more memory.

For methods relevant to this file, we can retain the same code as before from https://github.com/wavefrontHQ/go-metrics-wavefront/blob/master/wavefront.go

Reply:
is need it because the base registry ignore other counter types, like out Histogram, take a look:
https://github.com/rcrowley/go-metrics/blob/3113b8401b8a98917cde58f8bbd42a1b1c03b1fd/registry.go#L203

Reply:
Forget, now wf histogram is metrics.Histogram, so no need for custom registry.